data/conf/hive-site-old.xml

```diff
@@ -80,14 +80,6 @@
   <description></description>
 </property>
 
-<property>
-  <name>hive.metastore.metadb.dir</name>
```

Comment:
there plenty of places in a project with that config

Reply:
I have checked this with this PR and didn't find the obsolete  config. 
The hive-site.xml will be copied from `data/conf`  to other target places e.g `itest/qtest/target/llap` , therefore maybe you should `mvn clean install -DskipTests -Pitests -Piceberg` with the PR to clean original `hive-site.xml` in target directory, and you then won't find these obsolete configs.

Reply:
I compiled your PR, but still I can see this config in the target folders

Reply:
@ayushtkn It's weird. I compiled with the change again, and i still don't find the config.
Step to verify this PR:

1. Complie Hive project with this PR:
`mvn clean install -DskipTests -Pitests -Piceberg`

2. Search the obsolete config to make sure it was cleaned:

    `grep -rnw . -e "hive.metastore.metadb.dir"`

Reply:
@ayushtkn Thanks for your test. I test this PR again:

- Build the code against this PR:

`mvn clean install -DskipTests -Pitests -Piceberg`

- Search the obsolete config and the output as follows:

```
[INFO] Total time:  18:48 min
[INFO] Finished at: 2023-08-02T21:47:27+08:00
[INFO] ------------------------------------------------------------------------
[root@zbtdev hive]# grep -rnw . -e "hive.metastore.metadb.dir"
./shims/scheduler/target/testconf/hive-site-old.xml:84:  <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/hive-site.xml:128:  <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/iceberg/llap/hive-site.xml:119:        <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/iceberg/tez/hive-site.xml:119:  <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/llap/hive-site.xml:138:  <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/rlist/hive-site.xml:122:  <name>hive.metastore.metadb.dir</name>
./shims/scheduler/target/testconf/tez/hive-site.xml:114:  <name>hive.metastore.metadb.dir</name>
```

There is only `shims/scheduler` module has the obsolete config, but the module has been removed since HIVE-27131. That is to say, we can not clean the `shims/scheduler` target directory using` mvn clean` command. If it is necessary, i can remove the target directory of the module to avoid misleading.

But the interesting thing is that my output is different yours, I really don't know where the problem it is. Did you build the code with command which is same as mine: `mvn clean install -DskipTests -Pitests -Piceberg`?
 Could you please check it again? Many thanks.

Reply:
@zhangbutao I tried again. I have attached the output on the Jira
https://issues.apache.org/jira/secure/attachment/13061877/output.out

Reply:
yep, that works, let me circle back & check the PR again

Reply:
@ayushtkn Kindly ping, do you think we can proceed this change? :)